BootloaderCorePkg/Include/Library/BoardInitLib.h

```diff
@@ -118,6 +118,23 @@ PlatformUpdateHobInfo (
   OUT      VOID                  *HobInfo
   );
 
+/**
+  Return current version of the firmware requested.
+
+  This function will update and return the current
+  version of the firmware requested. If the firmware requested
+  is not supported by the platform, return 0 for version.
+
+  @param  Guid          The GUID to identify the firmware.
+  @param  Version       Pointer to the version information.
+
+**/
+VOID
+PlatformUpdateCurrentVersion (
+  IN CONST EFI_GUID   *Guid,
+  IN OUT UINT32       *Version
+  );

```

Comment:
Add EFIAPI modifier for lib API.
is it possible to use component signature instead of GUID for the first parameter?
In our internal flash map, we use signature to identify component, it will be consistent with flash map. And we will support more components to update, it would be better if we could avoid to add more and more guids for them.

Reply:
I will add EFIAPI.

Firmwares inside the capsule are identified with the GUID, this is according to the UEFI specification. It is unnecessary to add a conversion from GUID to name again.

Reply:
This API is used to get current version, not the version inside capsule image, right? 
if so, it is not related with GUID inside capsule image.

Reply:
we update the firmware from capsule  image, we identify using the GUID, to identify we need to declare a GUID, correct ? I reused this same GUID to update current version as well.

Do you suggest creating a Mapping this GUID to string again?

Reply:
we should separate "getting firmware version from current SBL" and "getting firmware version from capsule". For the first one, since there is GUID in current SBL to identify a component, so it is nature to use Signature instead of a GUID.
In the firmware update payload, currently we use a GUID to identify a component, so we could do convert in firmware update payload. Later, we might update capsule layout to directly use signature to identify a component in new capsule image. so we don't need change this API at that time.